scalableminds/webknossos
# Better error message when trying to upload annotation to missing dataset

When uploading an NML to a dataset that has no content (e.g. empty folder in binaryData dir) or is otherwise broken, the user now gets a readable error message rather than the generic Couldn’t find the requested resource.

Also, we save a few database requests by only constructing the dataSource once.

### Steps to test:
- Create empty dataset folder in binaryData dir, let wk scan the dir.
- Upload an annotation with `<experiment name="missing" />`
- Should get a readable error message
- Uploading to an existing working dataset should still work too.
